integration/truncindex_test.go

```diff
@@ -0,0 +1,133 @@
+	}()
+
+	t.Logf("Get image status by truncindex, truncId: %s", imgTruncId)
+	res, err := imageService.ImageStatus(&runtimeapi.ImageSpec{Image: imgTruncId})
+	require.NoError(t, err)
+	require.Equal(t, true, (res != nil) && (res.Id != ""))
+	t.Logf("Image status: %v", res)
+
+	// TODO(yanxuean): for failure test case where there are two images with the same truncindex.
+	// if you add n images at least two will share the same leading digit.
+	// "sha256:n" where n is the a number from 0-9 where two images have the same trunc,
+	// for example sha256:9
+	// https://github.com/kubernetes-incubator/cri-containerd/pull/352
+	// I am thinking how I get the two image which have same trunc.
+
+	// TODO(yanxuean): add test case for ListImages
+
+	t.Logf("Create a sandbox")
+	sbConfig := PodSandboxConfig("sandbox", "TruncIndex")
+	sb, err := runtimeService.RunPodSandbox(sbConfig)
+	require.NoError(t, err)
+	sbTruncIndex := genTruncIndex(sb)
+	defer func() {
+		assert.NoError(t, runtimeService.StopPodSandbox(sbTruncIndex))
+		assert.NoError(t, runtimeService.RemovePodSandbox(sbTruncIndex))
+	}()
+
+	t.Logf("Get sandbox status by truncindex")
+	_, err = runtimeService.PodSandboxStatus(sbTruncIndex)
+	require.NoError(t, err)
+
+	t.Logf("Forward port for sandbox by truncindex")
+	_, err = runtimeService.PortForward(&runtimeapi.PortForwardRequest{sbTruncIndex, []int32{80}})
+	require.NoError(t, err)
+
+	// TODO(yanxuean): add test case for ListPodSandbox
+
+	t.Logf("Create a container")
+	cnConfig := ContainerConfig(
+		"containerTruncIndex",
+		appImage,
+		WithCommand("top"),
+	)
+	cn, err := runtimeService.CreateContainer(sbTruncIndex, cnConfig, sbConfig)
+	require.NoError(t, err)
+	cnTruncIndex := genTruncIndex(cn)
+	defer func() {
+		assert.NoError(t, runtimeService.RemoveContainer(cnTruncIndex))
+	}()
+
+	t.Logf("Get container status by truncindex")
+	_, err = runtimeService.ContainerStatus(cnTruncIndex)
+	require.NoError(t, err)
+
+	t.Logf("Start the container")
+	require.NoError(t, runtimeService.StartContainer(cnTruncIndex))
+	defer func() {
+		assert.NoError(t, runtimeService.StopContainer(cnTruncIndex, 10))
+	}()
+
+	t.Logf("Stats the container")
+	_, err = runtimeService.ContainerStats(cnTruncIndex)
+	require.NoError(t, err)
+
+	t.Logf("Update container memory limit after started")
+	err = runtimeService.UpdateContainerResources(cnTruncIndex, &runtimeapi.LinuxContainerResources{
+		MemoryLimitInBytes: 50 * 1024 * 1024,
+	})
+	require.NoError(t, err)
+
+	t.Logf("Execute cmd in container")
+	execReq := &runtimeapi.ExecRequest{
+		ContainerId: cnTruncIndex,
+		Cmd:         []string{"pwd"},
+		Stdout:      true,
+	}
+	_, err = runtimeService.Exec(execReq)
+	require.NoError(t, err)
+
+	t.Logf("Execute cmd in container by sync")
+	_, _, err = runtimeService.ExecSync(cnTruncIndex, []string{"pwd"}, 10)
+	require.NoError(t, err)
+
+	// TODO(yanxuean): add test case for ListContainers
```

Comment:
Just stop and delete current sandbox, and get it again.

Reply:
There are defer functioin in first RunPodSandbox. If we explicitly stop and remove sandbox, the defer will fail
```
defer func() {
		assert.NoError(t, runtimeService.StopPodSandbox(sbTruncIndex))
		assert.NoError(t, runtimeService.RemovePodSandbox(sbTruncIndex))
	}()
```

Reply:
You can remove the assertion above. That is only for cleanup. In normal test run, it should get here and stop/remove.
You can add assertion here. ;)

If you create a new pod sandbox, you don't even know whether the trunc index has conflict with existing pod sandbox or not, because you haven't tried. So it's still possible that `runtimeService.PodSandboxStatus(sbTruncIndex1)` returns no error.

This is a little bit picky. It's up to you. :)

Reply:
done